src/utils/result-list.js

```diff
@@ -0,0 +1,56 @@
+const { findDOMNode } = require("react-dom");
+
+function scrollList(elem, index) {
+  const resultsEl = findDOMNode(elem).querySelector(".result-list");
+  if (!resultsEl || resultsEl.children.length === 0) {
+    return;
+  }
+
+  const resultsHeight = resultsEl.clientHeight;
+  const itemHeight = resultsEl.children[0].clientHeight;
+  const numVisible = resultsHeight / itemHeight;
+  const positionsToScroll = index - numVisible + 1;
+  const itemOffset = resultsHeight % itemHeight;
+  const scroll = positionsToScroll * (itemHeight + 2) + itemOffset;
+
+  resultsEl.scrollTop = Math.max(0, scroll);
+}
+
+function handleKeyDown(e: SyntheticKeyboardEvent) {
```

Comment:
I want to see if I can break this  out of `Autocomplete` and `SearchBar` so it's easily shared.